src/portable/nuvoton/nuc121/dcd_nuc121.c

```diff
@@ -368,13 +381,15 @@ void dcd_edpt_clear_stall(uint8_t rhport, uint8_t ep_addr)
   (void) rhport;
   USBD_EP_T *ep = ep_entry(ep_addr, false);
   ep->CFG |= USBD_CFG_CSTALL_Msk;
+  ep->CFG &=~USBD_CFG_DSQSYNC_Msk;
```

Comment:
we should clear the Data PID first before clear stall to prevent possible race condition. Could you move this before the clear stall.

Reply:
I have pushed update to correct this.